executor/aggregate_xapi.go

```diff
@@ -265,3 +265,139 @@ func needValue(name string) bool {
 	return name == ast.AggFuncSum || name == ast.AggFuncAvg || name == ast.AggFuncFirstRow ||
 		name == ast.AggFuncMax || name == ast.AggFuncMin || name == ast.AggFuncGroupConcat
 }
+
+// NewXAggregateExec deals with all the aggregate functions.
+// It is built from Aggregate Plan. When Next() is called, it reads all the data from Src and updates all the items in AggFuncs.
+type NewXAggregateExec struct {
+	Src               Executor
+	AggFields         []*types.FieldType
+	executed          bool
+	ctx               context.Context
+	AggFuncs          []expression.AggregationFunction
+	aggregaters       []*finalAggregater
+	groupMap          map[string]bool
+	groups            []string
```

Comment:
Because aggregate context use string as group key. I will refactor it in the next PR.